### PR TITLE
os/filestore/FileJournal: set block size via config option

### PR DESCRIPTION
**Pull request by liewegas**
We were setting the block_size as the MIN of the min size (4096) and the
block size reported for the journal file/device.  In reality, file systmes
report all kinds of crazy block sizes.  Usually it's the page size, but
sometimes it is larger (e.g., 128KB for ZFS), and that's not actually what
we want.  Using a size smaller than the file systems block size is not
optimal for performance, but it doesn't affect how the IO happens--as long
as it is larger than the hardware sector size, which is either 512 or
4096 bytes.  And our min was hard-coded at 4096.

So, instead, just set a config option to specify teh block size, and
default that to 4096.

The other uses of this constant we about *alignment* of memory buffers
for the purposes of direct IO.  Rename the option but do not change
the logic.  That means we continue to use 4k alignment for direct io even
if the device has 512 byte sectors, but that's fine--no reason to use
a smaller alignment.

Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

**Comment by wjwithagen** [comment]
@liewegas 
I think that the line with 
    assert(0 == "bl should be align");
Should move to the bottom of the code block.
Since this assert always triggers, the others in the block are irrelevant.
Changing the order allows the other statements to execute at least their tests.
If nothing trigger we get this line that triggers.

**Comment by liewegas** [comment]
@wjwithagen thanks, fixed the assert placement.

File: src/os/filestore/FileJournal.cc [hunk 1]
```diff
@@ -1038,13 +1041,13 @@ void FileJournal::align_bl(off64_t pos, bufferlist& bl)
...
-	(pos & (CEPH_MINIMUM_BLOCK_SIZE - 1)) != 0)
+		   !bl.is_n_align_sized(CEPH_DIRECTIO_ALIGNMENT))) {
+    if ((bl.length() & (CEPH_DIRECTIO_ALIGNMENT - 1)) != 0 ||
+	(pos & (CEPH_DIRECTIO_ALIGNMENT - 1)) != 0)
       dout(0) << "rebuild_page_aligned failed, " << bl << dendl;
-    assert((bl.length() & (CEPH_MINIMUM_BLOCK_SIZE - 1)) == 0);
-    assert((pos & (CEPH_MINIMUM_BLOCK_SIZE - 1)) == 0);
+    assert((bl.length() & (CEPH_DIRECTIO_ALIGNMENT - 1)) == 0);
+    assert((pos & (CEPH_DIRECTIO_ALIGNMENT - 1)) == 0);
+    assert(0 == "bl should be align");
```
Comment (dmick) [inline src/os/filestore/FileJournal.cc]: "aligned", while we're at it, and I think it's clearer to say "bl is not aligned"

**Comment by wjwithagen** [comment]
looks good to me.
Testing went from 27 FAIL to 20 FAIL.
So 7 more osd tests running to completion.
And the align assert has NOT occurred at all

**Pull request merged**